internal/dashboard/business/oceanbase/obcluster_usage.go

```diff
@@ -0,0 +1,119 @@
+	obcluster, err := oceanbase.GetOBCluster(ctx, nn.Namespace, nn.Name)
+	if err != nil {
+		return nil, err
+	}
+	clt := client.GetClient()
+	serverList := &v1alpha1.OBServerList{}
+	err = oceanbase.ServerClient.List(ctx, nn.Namespace, serverList, metav1.ListOptions{})
+	if err != nil {
+		return nil, httpErr.NewInternal(err.Error())
+	}
+	rootSecret, err := clt.ClientSet.CoreV1().Secrets(nn.Namespace).Get(ctx, obcluster.Spec.UserSecrets.Root, metav1.GetOptions{})
+	if err != nil {
+		return nil, httpErr.NewInternal(err.Error())
+	}
+	password, ok := rootSecret.Data["password"]
+	if !ok {
+		return nil, httpErr.NewInternal("root password not found")
+	}
+	var manager *operation.OceanbaseOperationManager
+	for _, observer := range serverList.Items {
+		source := connector.NewOceanBaseDataSource(observer.Status.GetConnectAddr(), oceanbaseconst.SqlPort, "root", "sys", string(password), oceanbaseconst.DefaultDatabase)
+		manager, err = operation.GetOceanbaseOperationManager(source)
+		if err == nil {
+			break
+		}
+	}
+	if manager == nil {
+		return nil, httpErr.NewInternal("no running observer is connectable")
+	}
+	defer manager.Close()
+
+	parameters, err := manager.GetParameter("__min_full_resource_pool_memory", nil)
+	if err != nil {
+		return nil, httpErr.NewInternal(err.Error())
+	}
+	minPoolMemory, err := resource.ParseQuantity(parameters[0].Value)
+	if err != nil {
+		return nil, httpErr.NewInternal(err.Error())
+	}
+	essentials := &response.OBClusterEssentialParameters{
+		MinPoolMemory: minPoolMemory.Value(),
+	}
+	gvservers, err := manager.ListGVServers()
+	if err != nil {
+		return nil, httpErr.NewInternal(err.Error())
+	}
+	serverUsages, zoneMapping := getServerUsages(gvservers)
+	essentials.OBServerResources = serverUsages
+	essentials.OBZoneResourceMap = zoneMapping
+	return essentials, nil
+}
+
+func getServerUsages(gvservers []model.GVOBServer) ([]response.OBServerAvailableResource, map[string]*response.OBZoneAvaiableResource) {
+	zoneMapping := make(map[string]*response.OBZoneAvaiableResource)
+	serverUsages := make([]response.OBServerAvailableResource, 0, len(gvservers))
+	for _, gvserver := range gvservers {
+		zoneResource := &response.OBZoneAvaiableResource{
+			OBZone:            gvserver.Zone,
+			AvailableCPU:      gvserver.CPUCapacity - gvserver.CPUAssigned,
+			AvailableMemory:   gvserver.MemCapacity - gvserver.MemAssigned,
+			AvailableLogDisk:  gvserver.LogDiskCapacity - gvserver.LogDiskAssigned,
+			AvailableDataDisk: gvserver.DataDiskCapacity - gvserver.DataDiskAllocated,
+		}
+		if _, ok := zoneMapping[gvserver.Zone]; !ok {
+			zoneMapping[gvserver.Zone] = zoneResource
+		}
+		serverUsage := response.OBServerAvailableResource{
+			OBServerIP:             gvserver.ServerIP,
+			OBZoneAvaiableResource: *zoneResource,
+		}
+		if zoneMapping[gvserver.Zone].AvailableCPU > serverUsage.AvailableCPU {
```

Comment:
why using the minimum value

Reply:
It means that all servers in the zone at least have that much resources.

Reply:
OK, I set it to the maximum available value